bin/ch/Helpers.cpp

```diff
@@ -4,6 +4,212 @@
 //-------------------------------------------------------------------------------------------------------
 #include "stdafx.h"
 
+//TODO: x-plat definitions
+#ifdef _WIN32
+typedef char16 TTDHostCharType;
+typedef struct _wfinddata_t TTDHostFileInfo;
+typedef intptr_t TTDHostFindHandle;
+typedef struct _stat TTDHostStatType;
+
+#define TTDHostPathSeparator _u("\\")
+#define TTDHostPathSeparatorChar _u('\\')
+#define TTDHostFindInvalid -1
+
+size_t TTDHostStringLength(const TTDHostCharType* str)
+{
+    return wcslen(str);
+}
+
+void TTDHostInitEmpty(TTDHostCharType* dst)
+{
+    dst[0] = _u('\0');
+}
+
+void TTDHostInitFromUriBytes(TTDHostCharType* dst, const byte* uriBytes, size_t uriBytesLength)
```

Comment:
This design allows the host to represent the URI in whatever format it likes (wchar, utf8, etc.) and allows ChakraCore to just pass the buffer of bytes without needing to know anything about them. Thus, the CH host can use char16 for windows and utf8 on posix. I thought about using a single char* representation in the Host but I didn't like having the utf8/char16 conversions anytime I accessed a file/directory.